### PR TITLE
Scaffold index view render

### PR DESCRIPTION
**Pull request by Otterpocket**
### Summary

Currently when you use scaffold it creates an index view that uses '@cats.each' on the collection rather than using 
<%= render @cats %>
with a partial _cat.html.erb. 
I see this as a good opportunity for new users of the framework to learn the collection render feature.

This commit updates the scaffold generator to create the member partial and modifies the index view to use render instead of each.

### CODE REVIEW

**Comment by utilum** [comment]
It seems sensible. Also seems that if adopted it should be applied consistently, to any `:index` or collection view generator.

File: railties/lib/rails/generators/erb/scaffold/scaffold_generator.rb [hunk 1]
```diff
@@ -15,6 +15,7 @@ def create_root_folder
       end
 
       def copy_view_files
+        template "_member_partial.html.erb", File.join("app/views", controller_file_path, "_#{file_name}.html.erb")
```
Comment (rafaelfranca) [inline railties/lib/rails/generators/erb/scaffold/scaffold_generator.rb]: Can we add the view to the `available_views` array?

**Review by rafaelfranca — COMMENTED** [review]
I'm on the fence on this. @dhh can you take a look?

**Comment by dhh** [comment]
I sympathize with going for this, but I think it introduces indirection that isn't going to help with initial uptake. I'd leave things as they are.

**Pull request closed without merging**